blivet/dbus/blivet.py

```diff
@@ -100,9 +100,13 @@ def ResolveDevice(self, spec):
         """ Return a string describing the device the given specifier resolves to. """
         device = self._blivet.devicetree.resolve_device(spec)
         object_path = ""
-        if device is not None:
-            dbus_device = next(d for d in self._dbus_devices if d._device == device)
-            object_path = dbus_device.object_path
+        if device is None:
+            raise dbus.exceptions.DBusException('%s.DeviceLookupFailed' % BUS_NAME,
+                                                'No device was found that matches the device '
+                                                'descriptor "%s".' % spec)
+
+        dbus_device = next(d for d in self._dbus_devices if d._device == device)
+        object_path = dbus_device.object_path
```

Comment:
Either an exception or ``'/'`` as that's what other DBus APIs do in cases like this. On the other hand I think returning ``'/'`` would make more sense if the method was called like ``FindDevice()``. ``ResolveDevice()`` can consider the device not existing an exception.